CMakeLists.txt

```diff
@@ -163,5 +166,22 @@ if(SOCI_TESTS)
   add_subdirectory(tests)
 endif()
 
+###############################################################################
+# build config file
+###############################################################################
```

Comment:
You both present no very strong valid points. I have no strong opinion but since we already have been hit by bug due to missed maintenance of such definitions I lean towards @ArnaudD-FR 's auto mode. But I'd like to see

1. Clear naming convention E. g. `SOCI_HAVE_*` only for generated configuration variables.

2. Way to dump or list all such variables automatically so it is easy to write them on docs without searching through sources, missing any.

I can write docs in markdown for them.